xla/python/profiler.cc

```diff
@@ -270,22 +278,28 @@ void BuildProfilerSubmodule(nb::module_& m) {
       },
       nb::arg("tensorboard_dir"));
 
-  profiler.def(
-      "get_instructions_profile",
-      [](const std::string& tensorboard_dir)
-          -> std::vector<std::pair<std::string, double>> {
-        tensorflow::profiler::ProfiledInstructionsProto profile_proto;
-        xla::ThrowIfError(
-            xla::ConvertXplaneUnderLogdirToProfiledInstructionsProto(
-                tensorboard_dir, &profile_proto));
-        std::vector<std::pair<std::string, double>> results;
-        results.reserve(profile_proto.costs().size());
-        for (const auto& c : profile_proto.costs()) {
-          results.emplace_back(c.name(), c.cost_us());
-        }
-        return results;
-      },
-      nb::arg("tensorboard_dir"));
+  // The difference between this API and `get_fdo_profile` is that this
+  // API parses the protos to in-memory Python objects so that they can be
+  // easily manipulated by the users.
+  nb::class_<ProfiledInstruction> profiled_instruction_class(
+      profiler, "ProfiledInstruction");
+  profiled_instruction_class
+      .def_ro("name", &ProfiledInstruction::instruction_name)
+      .def_ro("cost_us", &ProfiledInstruction::cost_us);
+
+  profiler.def("parse_profiled_instructions_proto",
+               [](nb::bytes profile) -> std::vector<ProfiledInstruction> {
+                 tensorflow::profiler::ProfiledInstructionsProto profile_proto;
+                 profile_proto.ParseFromString(
+                     std::string(profile.c_str(), profile.size()));
+
+                 std::vector<ProfiledInstruction> result;
+                 for (const auto& c : profile_proto.costs()) {
+                   result.push_back(ProfiledInstruction{
+                       .instruction_name = c.name(), .cost_us = c.cost_us()});
```

Comment:
Designated initializers are incompatible with C++ standards before C++20.
Please avoid them in the OpenXLA repo at the moment.